src/Dropbox.elm

```diff
@@ -861,7 +924,148 @@ upload auth info =
+                    ]
+                )
+            )
+        |> Pipeline.required "cursor" Json.Decode.string
+        |> Pipeline.required "has_more" Json.Decode.bool
+
+
+{-| Begin listing the contents of a folder.
+
+See <https://www.dropbox.com/developers/documentation/http/documentation#files-list_folder>
+
+-}
+listFolder : UserAuth -> ListFolderRequest -> Task ListFolderError ListFolderResponse
+listFolder auth options =
+    let
+        url =
+            "https://api.dropboxapi.com/2/files/list_folder"
+
+        body =
+            Json.Encode.encode 0 <|
+                Json.Encode.object <|
+                    [ ( "path", Json.Encode.string options.path )
+                    , ( "recursive", Json.Encode.bool options.recursive )
+                    , ( "include_media_info", Json.Encode.bool options.includeMediaInfo )
+                    , ( "include_deleted", Json.Encode.bool options.includeDeleted )
+                    , ( "include_has_explicit_shared_members", Json.Encode.bool options.includeHasExplicitSharedMembers )
+                    ]
+
+        decodeError err =
+            case err of
+                Http.BadStatus response ->
+                    case Json.Decode.decodeString decodeListError response.body of
+                        Ok err ->
+                            err
+
+                        Err _ ->
+                            OtherListFailure err
+
+                _ ->
+                    OtherListFailure err
+    in
+    Http.request
+        { method = "POST"
+        , headers =
+            [ authHeader auth ]
+        , url = url
+        , body = Http.stringBody "application/json" body
+        , expect = Http.expectJson decodeListResponse
+        , timeout = Nothing
+        , withCredentials = False
+        }
+        |> Http.toTask
+        |> Task.mapError decodeError
+
+
+listFolderContinue : UserAuth -> ListFolderResponse -> Task ListFolderError ListFolderResponse
+listFolderContinue auth lastResult =
+    let
+        url =
+            "https://api.dropboxapi.com/2/files/list_folder/continue"
+
+        body =
+            Json.Encode.encode 0 <|
+                Json.Encode.object <|
+                    [ ( "cursor", Json.Encode.string lastResult.cursor )
+                    ]
+
+        decodeError err =
+            case err of
+                Http.BadStatus response ->
+                    case Json.Decode.decodeString decodeListError response.body of
+                        Ok err ->
+                            err
+
+                        Err _ ->
+                            OtherListFailure err
+
+                _ ->
+                    OtherListFailure err
+    in
+    Http.request
+        { method = "POST"
+        , headers =
+            [ authHeader auth ]
+        , url = url
+        , body = Http.stringBody "application/json" body
+        , expect = Http.expectJson decodeListResponse
```

Comment:
Lots of redundancy in the request generation. I considered refactoring this to shared code, but figured I'd get some feedback first.

A generic Dropbox request builder might even be useful to expose as API future-proofing and a work-around to provide access to unimplemented calls. Then again, maybe that would discourage folks from PR-ing new functionality.